wasm-chain-integration/src/v1/trie/low_level.rs

```diff
@@ -923,9 +925,10 @@ type Position = u16;
 
 #[derive(Debug)]
 pub struct Iterator {
-    /// The root of the iterator
+    /// The root and generation of the iterator
     /// This is stored here to allow efficient deleting of iterators.
-    root:         usize,
+    generation:   u32,
+    root:         Vec<u8>,
```

Comment:
Document why this is needed. 

Also I do not like duplicating the data here. When the btreemap is replaced by something else we'd ideally have this be a pointer in there.

With a btreemap this should probably be a `Rc<[u8]>`, and likewise in the btreemap we should store Rc<[u8]> instead of vec.